### PR TITLE
Support Rack 3

### PR DESCRIPTION
**Pull request by dentarg**
All builds pass: https://github.com/dentarg/rack-contrib/actions/runs/6196721481

Builds on https://github.com/rack/rack-contrib/pull/186

Close https://github.com/rack/rack-contrib/issues/183

### CODE REVIEW

**Review by ioquatix — APPROVED** [review]

**Comment by ioquatix** [comment]
I am totally fine with dropping old Ruby versions that are EOL if it makes things easier.

**Comment by ioquatix** [comment]
Can you rebase? Thanks!

**Comment by dentarg** [comment]
Rebase done!

File: lib/rack/contrib/common_cookies.rb [hunk 1]
```diff
@@ -13,7 +13,8 @@ def initialize(app)
 
     def call(env)
       status, headers, body = @app.call(env)
-      headers = Utils::HeaderHash.new(headers)
+      headers_klass = Rack.release < "3" ? Utils::HeaderHash : Headers
```
Comment (ioquatix) [inline lib/rack/contrib/common_cookies.rb:16]: What do you think about using a private const for this?

Comment (dentarg) [reply lib/rack/contrib/common_cookies.rb:16]: Sure, makes sense

Comment (dentarg) [reply lib/rack/contrib/common_cookies.rb]: 836bf7d

**Comment by dentarg** [comment]
Let me know if I can do anything more here :)

**Review by ioquatix — APPROVED** [review]

**Comment by dentarg** [comment]
@ioquatix are you able to cut a new release with this?

**Comment by ioquatix** [comment]
Released v2.4.0 - let me know if you have any issues.

**Comment by dentarg** [comment]
Thanks for the quick turnaround!

**Pull request merged**